services/checks/checks.go

```diff
@@ -131,13 +130,301 @@ func (s *Service) Run(ctx context.Context) {
 
 // Checks returns available checks.
 func (s *Service) Checks() []check.Check {
-	s.m.RLock()
-	defer s.m.RUnlock()
+	s.cm.Lock()
+	defer s.cm.Unlock()
 
 	r := make([]check.Check, 0, len(s.checks))
 	return append(r, s.checks...)
 }
 
+func (s *Service) processTasks(ctx context.Context, tasks []task) {
+	ticker := time.NewTicker(time.Minute)
```

Comment:
We should stop it in `defer`.

Reply:
And that value is too high: currently, all actions are limited to 10 seconds: https://github.com/percona/pmm-agent/blob/a2844d35b8e1a2fb25ef41f80d2258ab99a07721/client/client.go#L95

Reply:
Set same value, 10 seconds for result awaiting

Reply:
Changed it to 15 seconds to cascade timeouts